integration_tests/test_ledger_evm.py

```
@@ -0,0 +1,141 @@
+        return None
+
+    expected_sender = tx_params["from"].lower()
+    r = int.from_bytes(response[1:33], "big")
+    s = int.from_bytes(response[33:65], "big")
+    for recovery_id in [0, 1]:
+        eip155_v = recovery_id + (w3.eth.chain_id * 2) + 35
+        recovered = recover_sender(tx_hash, eip155_v, r, s)
+        if recovered.lower() == expected_sender:
+            break
+    else:
+        raise ValueError("recover sender fails")
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)